articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,340 @@
+
+> Note that the code block in the Do-While loop executes at least once.
+
+***Example:***
+
+```Dart
+void main(){
+   int i = 0;
+   do {
+      print ("Hello");
+      i++;
+   } while (i<5);
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+
+The code first executes then the counter variable increments by one. It then checks the condition (1<5). It is true, so the loop will begin again. The code executes, then the counter variable increments by one. It then checks the condition (2<5). It meets the condition. It will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.
+
+### Other Key Concepts
+#### Break Keyword
+
+We use the `Break Keyword` to forcefully end a loop. It comes in handy when you want a partial output on a certain condtion and not to run all iterations.
+
+***Example:***
+
+```Dart
+void main(){
+ for (int i = 1; i<=10; i++){
+    print ("Hello");
+    if(i >= 6){
+       break;
+    }
+  }
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+This code will end the loop after printing five hellos because of the break statement.
+
+#### Continue Keyword
+
+We use the `Continue Keyword` to skip code or a loop iteration after the continuation and proceed with the rest of the iteration.
+
+***Example:***
+
+```Dart
+void main(){
+
+ for (int i = 1; i<=8; i++){
+   
+    print (i);
+
+    if(i == 5){
+      
+       continue;
+    }
+
+  }
+   
+}
+```
+Output:
+
+```
+1
+2
+3
+4
+5
+6
+7
+8
+```
+In the example, when i becomes 5, it skips the print statement, the rest of the loop continues.
+
+### Applications of Loops
+Here are some positive and negative applications of loops:
+
+1. **Writing a Music Player Code**: A music player has a loop that enables it to play from one song to the next systematically.
+
+2. **Cycling through values**: It is the use of loops to print out values in an extensive list. ***For example***, Names of students in a school.
+
+```Dart
+void main(){
+   
+   List grade-one-students = ["John", "Mark", "Alex"];
+   
+   for(String student in grade-one-students){
+
+      print(student);
+
+   }
+}
+```
+Output:
+
+```
+John
+Mark
+Alex
+```
+
+It loops over all the elements stored in the list and prints them out.
+
+3. **Create computer viruses**: Malicious programmers use loops to create [Computer Viruses](https://en.wikipedia.org/wiki/Computer_virus) by using the While loop and embedding it to software downloads. If we write a while loop without the increment or decrement counter variable, it creates a never-ending loop which leads to unnecessary storage and memory consumption on devices. This courses the devices to crash or behave abnormally.
+
+***Caution: DO NOT RUN THIS CODE!!!***
+
+```Dart
+void main(){
+
+   int i = 0;
+
+   while(i<3){
+      print("Hello");
+      
+   }
+
+}
+```
+
+Output :
+
+``` 
+   Hello
+   Hello
+```
```

Comment:
You can remove this output, You've stated that it will print `Hello` endlessly, but you've still added this output section with 2 `Hello`.